Assets/Capstone/Scripts/Environmental/Sun.cs

```diff
@@ -46,69 +67,149 @@ void Update()
         //rotates sun around zero 
         transform.RotateAround(Vector3.zero, Vector3.forward, rotationSpeed * Time.deltaTime);
 
-        if(transform.position.x > middayInterval)
+        //always look at center of the map
+        transform.LookAt(Vector3.zero);
+
+        CheckSunsRotation();
+    }
+
+    void CheckSunsRotation()
+    {
+        //for checking angle
+        Vector3 forward = transform.forward;
+        angleInDegrees = Mathf.Atan2(forward.y, forward.x) * Mathf.Rad2Deg;
+
+        //is Morning 
+        if (angleInDegrees > morningInterval || angleInDegrees < middayInterval)

```

Comment:
I can't think of it right now but there is probably a more efficient way to do these calculations instead of chaining ifs like this.